lib/addressable/uri.rb

```diff
@@ -687,8 +687,7 @@ def self.normalized_encode(uri, return_type=String)
       components.each do |key, value|
         if value != nil
           begin
-            components[key] =
-              Addressable::IDNA.unicode_normalize_kc(value.to_str)
+            components[key] = value.to_str.unicode_normalize(:nfc)
```

Comment:
Same thing for the both calls in URI we should use NFC form instead of NFKC (only for IDNA), so I switched to the simple ruby call. `:nfc` is the default form but I let it as an argument to show that it's a deliberate choice.